### PR TITLE
Wire up geo_bounds aggregation to ValuesSourceRegistry

### PR DESCRIPTION
**Pull request by talevy**
This commit updates the geo_bounds aggregation to depend
on registering itself in the ValuesSourceRegistry

relates to additional unit tests for the GeoBoundsAggregator #53033.
relates to the values-source refactoring meta issue #42949.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsAggregatorSupplier.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.elasticsearch.search.internal.SearchContext;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public interface GeoBoundsAggregatorSupplier extends AggregatorSupplier {
+
+    GeoBoundsAggregator build(String name, SearchContext aggregationContext, Aggregator parent,
+                        ValuesSource.GeoPoint valuesSource, boolean wrapLongitude, List<PipelineAggregator> pipelineAggregators,
```
Comment (not-napoleon) [inline server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsAggregatorSupplier.java]: This should just accept a base `ValuesSource` and the implementation can cast it as needed.  That way, when we add a new `ValuesSource` subclass, we can still just pass it in here.

**Review by not-napoleon — CHANGES_REQUESTED** [review]
As per comment, the supplier signature needs to be tweaked.  Otherwise, it looks good.  Thanks!

File: server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsAggregatorSupplier.java (hunk 1 above)
Comment (talevy) [reply server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsAggregatorSupplier.java]: 👍

Comment (talevy) [reply server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsAggregatorSupplier.java]: actually. question about this. Would this mean that it is best if the Aggregator constructors abstract the constructor argument for this as well and cast internally?

**Review by not-napoleon — APPROVED** [review]
LGTM

**Pull request merged**